### PR TITLE
Recurring contributions: add to Collective page, use Graphql v2 for tiers

### PR DESCRIPTION
**Pull request by sbinlondon**
* Updates to use GraphQL v2 for the tier and amount updates
* Starts to add the recurring contributions to the Collective page
* Further work to make the standalone page look nicer (adding navbar)

Work in progress (i.e. I know there's work to be done on the StyledFilters on the Collective section)

### CODE REVIEW

File: components/collective-page/_constants.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ export const Sections = {
   CONVERSATIONS: 'conversations',
   TRANSACTIONS: 'transactions',
   ABOUT: 'about',
+  RECURRINGCONTRIBUTIONS: 'recurringcontributions',
```
Comment (Betree) [inline components/collective-page/_constants.js:32]: **[:ice_cream: Details & personal preferences]** 

```suggestion
  RECURRING_CONTRIBUTIONS: 'recurring-contributions',
```

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

File: components/collective-page/sections/RecurringContributions.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+import I18nFormatters from '../../I18nFormatters';
+import LoadingPlaceholder from '../../LoadingPlaceholder';
+import MessageBox from '../../MessageBox';
+import RecurringContributionsContainer from '../../recurring-contributions/RecurringContributionsContainer';
+import StyledFilters from '../../StyledFilters';
+import TemporaryNotification from '../../TemporaryNotification';
+import { P } from '../../Text';
+import { withUser } from '../../UserProvider';
+import ContainerSectionContent from '../ContainerSectionContent';
+// Local imports
```
Comment (Betree) [inline components/collective-page/sections/RecurringContributions.js:20]:
```suggestion
```

File: components/recurring-contributions/RecurringContributionsContainer.js [hunk 3]
```diff
@@ -42,25 +39,27 @@ const RecurringContributionsContainer = ({ recurringContributions, filter, creat
...
-            account={account}
-            data-cy="recurring-contribution-card"
-          />
-        </CollectiveCardContainer>
-      ))}
-    </AllCardsContainer>
+    <Container maxWidth={Dimensions.MAX_SECTION_WIDTH} pl={Dimensions.PADDING_X} mt={4} mx="auto">
+      <Flex flexWrap="wrap" justifyContent={['space-evenly', 'left']} my={2}>
+        {displayedRecurringContributions.map(contribution => (
+          <CollectiveCardContainer key={`${contribution.id}-container`}>
```
Comment (Betree) [inline components/recurring-contributions/RecurringContributionsContainer.js:45]: **[:ice_cream: Details & personal preferences]** Adding this suffix would be great for `data-cy`, but `key` only cares about uniqueness (it's not used elsewhere).

```suggestion
          <CollectiveCardContainer key={contribution.id}>
```

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

File: components/recurring-contributions/RecurringContributionsContainer.js [hunk 4]
```diff
@@ -42,25 +39,27 @@ const RecurringContributionsContainer = ({ recurringContributions, filter, creat
...
+            <RecurringContributionsCard
+              key={contribution.id}
```
Comment (Betree) [inline components/recurring-contributions/RecurringContributionsContainer.js:47]: No need to `key` this one

```suggestion
```

File: components/recurring-contributions/UpdateOrderPopUp.js [hunk 5]
```diff
@@ -120,12 +129,12 @@ const UpdateOrderPopUp = ({ setMenuState, contribution, createNotification, setS
...
+        amount: tier.amountType === 'FLEXIBLE' ? tier.minimumAmount.valueInCents : tier.amount.valueInCents,
         id: tier.id,
-        currency: tier.currency,
+        currency: tier.amount.currency,
         interval: tier.interval,
         presets: tier.presets,
-        minimumAmount: tier.amountType === 'FLEXIBLE' ? tier.minimumAmount : 100,
+        minimumAmount: tier.amountType === 'FLEXIBLE' ? tier.minimumAmount.valueInCents : 100,
       }));
     tierOptions.unshift(customTierOption);
```
Comment (Betree) [inline components/recurring-contributions/UpdateOrderPopUp.js:139]: **[:hammer_and_wrench: **Important**]** It's not from this PR, but I just realized we need to make sure that `collective.settings.disableCustomContributions` is not true before adding the "custom tier" option to the list (see https://github.com/opencollective/opencollective-api/pull/3936).

Can be done in another PR if more convenient.

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

Comment (sbinlondon) [reply components/recurring-contributions/UpdateOrderPopUp.js:139]: Good shout, I didn't realise that was a thing!

**Comment by sbinlondon** [comment]
@Betree The recurring contributions section comes after the About section when I view it on my profile page... do you know where I would specify the order of the sections?

**Comment by Betree** [comment]
@sbinlondon the default order is defined by the order of the sections in https://github.com/opencollective/opencollective-frontend/blob/d4aea3478bed35f7536f61a93bfca212eb668db8/components/CollectiveNavbar.js#L171

File: components/recurring-contributions/UpdateOrderPopUp.js (hunk 5 above)
Comment (sbinlondon) [reply components/recurring-contributions/UpdateOrderPopUp.js:139]: No big deal, added to this PR :)

**Pull request merged**